101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+        "Basic",
+        "Standard",
+        "Premium"
+      ]
+    },
+    "containerRegistryResourceGroupName": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]"
+    },
+    "containerRegistryBehindVNet": {
+      "type": "string",
+      "defaultValue": "false",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "vnetOption": {
+      "type": "string",
+      "defaultValue": "none",
+      "allowedValues": [
+        "new",
+        "existing",
+        "none"
+      ]
+    },
+    "vnetName": {
+      "type": "string",
+      "defaultValue": ""
+    },
+    "vnetResourceGroupName": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]"
+    },
+    "addressPrefixes": {
+      "type": "array",
+      "defaultValue": [
+        "10.0.0.0/16"
+      ],
+      "metadata": {
+        "description": "Address prefix of the virtual network"
+      }
+    },
+    "subnetOption": {
+      "type": "string",
+      "defaultValue": "none",
+      "allowedValues": [
+        "new",
+        "existing",
+        "none"
+      ]
+    },
+    "subnetName": {
+      "type": "string",
+      "defaultValue": ""
+    },
+    "subnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "Subnet prefix of the virtual network"
+      }
+    },
+    "adbWorkspace": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Azure Databrick workspace to be linked to the workspace"
+      }
+    },
+    "confidential_data":{
+      "type": "string",
+      "defaultValue": "false",
+      "allowedValues": [
+        "false",
+        "true"
+      ],
+      "metadata": {
+        "description": "Specifies that the Azure Machine Learning workspace holds highly confidential data."
+      }
+    },
+    "encryption_status":{
+      "type": "string",
+      "defaultValue": "Disabled",
+      "allowedValues": [
+        "Enabled",
+        "Disabled"
+      ],
+      "metadata": {
+        "description": "Specifies if the Azure Machine Learning workspace should be encrypted with customer managed key."
+      }
+    },
+    "cmk_keyvault":{
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Specifies the customer managed keyVault arm id."
+      }
+    },
+    "resource_cmk_uri":{
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Specifies if the customer managed keyvault key uri."
+      }
+    },
+    "privateEndpointType": {
+      "type": "string",
+      "defaultValue": "none",
+      "allowedValues": [
+        "AutoApproval",
+        "ManualApproval",
+        "none"
+      ]
+    }
+
+  },
+  "variables": {
+    "storageAccountName": "[if(empty(parameters('storageAccountName')), concat('sa',uniqueString(resourceGroup().id, parameters('location'))), parameters('storageAccountName'))]",
```

Comment:
revert this change (remove all the if() stmts) and use the defaultValues on the parameters